### DISCUSSION TITLE
[@types/react] There is no way to simply annotate function component with no props but with children.

### DISCUSSION THREAD

**Discussion by jakubmazanec** [DefinitelyTyped/DefinitelyTyped#60624] [Issues with a @types package] [answered]
AFAIK after release of @types/react v18, there is no simple way to annotate a function component that has no props, but can have children, e.g.:

```js
import {FC, PropsWithChildren} from 'react';

// could be shorter
let Bold: React.FC<PropsWithChildren<unknown>> = ({children}) => <b>{children}</b>;
```

At least, `PropsWithChildren` should have a default for its type parameter:

```js
import {FC, PropsWithChildren} from 'react';

let Bold: React.FC<PropsWithChildren> = ({children}) => <b>{children}</b>;
```

But better solution would be a new type replacing previous (v17) functionality of `FC` type:

```js
import {FCC} from 'react'; // `FCC` is short for `FunctionComponentWithChildren`

// nice...
let Bold: React.FCC = ({children}) => <b>{children}</b>;

// ...and short
interface FooProps {
  foo: string;
}

let Foo: React.FCC<FooProps> = ({foo, children}) => /* ... */
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
You're free to add these types to your codebase. But we want to reduce the number of initial types to make onboarding easier and reduce potential discussions (which type to use for a component with just children).

**Reply by jakubmazanec** [reply]
That's the point, there is no type to use for a component with just a children.

But in any case, changing `PropsWithChildren` to `type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined };` doesn't increase "number of initial types".

**Reply by eps1lon** [reply]
> But in any case, changing PropsWithChildren to type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined }; doesn't increase "number of initial types".

We had this discussion before and my argument was then that `PropsWithChildren` is most of the time a mistake: You forgot to pass in the props. I think it makes sense now since we can use `unknown` as the default not `{}`. With `{}` we would've accepted excess props. With `unknown` we don't so you see if `PropsWithChildren` was accidental or intentional.

So I suggest filing a PR with `PropsWithChildren<P = unknown>` and extensive tests for how excess props and props inference behaves.
